app/src/main/java/com/p2p/presentation/truco/TrucoViewModel.kt

```diff
@@ -391,28 +428,53 @@ abstract class TrucoViewModel(
                     currentActionPoints = 1
                 }
             }
+            is GoToDeck -> {
+                if (allPlayersHaveGoneToDeck(performedByTeam))
+                    looseRound(performedByTeam)
+                else playerGoToDeck(action.player)
+            }
         }
     }
 
+    private fun allPlayersHaveGoneToDeck(performedByTeam: Int): Boolean {
+        val myTeam = playersTeams.filter { it.team == performedByTeam }.map { it.player }
+        val myTeamInRound = cardsByPlayer.filter {
+            myTeam.contains(it.player) && it.cards.sumBy { it.number } != 0
+        }.size
+        return myTeamInRound - 1 == 0
+    }
+
+    private fun looseRound(
+        performedByTeam: Int,
+        winner: Int = playersTeams.first { it.team != performedByTeam }.team
+    ) {
+        onHandFinished(winner)
+    }
```

Comment:
Es medio rari este método porque con usando un parámetro por default estás arruinando toda la semántica. Está bueno, pero le sacaría el parámetro por default. Además, estás hablando de la mano, no de la ronda.
```suggestion
    private fun loseHand(loserTeam: Int) {
        onHandFinished(playersTeams.first { it.team != loserTeam }.team)
    }
```